src/main.rs

```diff
@@ -206,11 +208,21 @@ impl App {
             self.reply(Response::new_ok(id, document_links));
         } else if let Some((id, params)) = cast::<Formatting>(&mut req) {
             let changes = if let Some((ast, code, _)) = self.files.get(&params.text_document.uri) {
-                let fmt = nixpkgs_fmt::reformat_node(&ast.node());
-                vec![TextEdit {
-                    range: utils::range(&code, TextRange::up_to(ast.node().text().len())),
-                    new_text: fmt.text().to_string(),
-                }]
+                let (spacing_edits, indent_edits) = nixpkgs_fmt::reformat_edits(&ast.node());
+                let merged_edits = textedit_merge::merge(
```

Comment:
I'm not sure why this is needed, but I guess this can be answered with https://github.com/nix-community/nixpkgs-fmt/pull/296#discussion_r907196743

Reply:
Yeah, like I mentioned in https://github.com/nix-community/nixpkgs-fmt/pull/296#discussion_r907966988, the delete ranges in the spacing edits apply to the original document, and the indent edits apply to the document after the spacing edits have been applied. Since the language server protocol expects a single set of edits in response to a reformat request, I believe merging the two sequential sets of edits into a single set of edits is necessary.